knowledge-base/grid-show-confirm-cancel-dialog.md

```diff
@@ -0,0 +1,236 @@
+* Wait for the user choice to prevent or proceed with the cancel operation.
+* Hide the Dialog.
+
+````CSTHML
+@*Only OnCancel event is handled for brevity*@
+
+@using System.ComponentModel.DataAnnotations @* for the validation attributes *@
+
+<TelerikGrid Data=@GridData
+             EditMode="@GridEditMode.Inline"
+             Pageable="true"
+             Sortable="true"
+             Height="500px"
+             OnCancel="@CancelHandler"
+             ShowColumnMenu="true">
+    <GridToolBar>
+        <GridCommandButton Command="Add" Icon="add">Add Employee</GridCommandButton>
+    </GridToolBar>
+    <GridColumns>
+        <GridColumn Field=@nameof(SampleData.ID) Title="ID" Editable="false" />
+        <GridColumn Field=@nameof(SampleData.Name) Title="Name" />
+        <GridCommandColumn>
+            <GridCommandButton Command="Save" Icon="save" ShowInEdit="true">Update</GridCommandButton>
+            <GridCommandButton Command="Edit" Icon="edit">Edit</GridCommandButton>
+            <GridCommandButton Command="Cancel" Icon="cancel" ShowInEdit="true">Cancel</GridCommandButton>
+        </GridCommandColumn>
+    </GridColumns>
+</TelerikGrid>
+
+<TelerikDialog @bind-Visible="@DialogVisible"
+               Title="@Title">
+    <DialogContent>
+        Are you sure you want to cancel the editing?
+    </DialogContent>
+    <DialogButtons>
+        <TelerikButton OnClick="@(() => CancelConfirmation("continueEdit"))">No, continue edit</TelerikButton>
+        <TelerikButton OnClick="@(() => CancelConfirmation("cancelEdit"))">Yes, cancel edit</TelerikButton>
+    </DialogButtons>
+</TelerikDialog>
+
+@code {
+    private List<SampleData> GridData { get; set; }
+
+    private bool DialogVisible { get; set; }
+
+    private string Title { get; set; } = "Please confirm";
+
+    private bool? ContinueEdit { get; set; }
+
+    private async Task CancelHandler(GridCommandEventArgs args)
+    {
+        SampleData item = (SampleData)args.Item;
+
+        //clean the value, so you can always get the updated one
+        ContinueEdit = null;
+
+        //show dialog
+        DialogVisible = true;
+
+        //wait for the user choice from the dialog handled in CancelConfirmation()
+        while (!ContinueEdit.HasValue)
+        {
+            await Task.Delay(10);
+        }
+
+        //cancel the event if the user wants to continue editing
+        args.IsCancelled = ContinueEdit.Value;
+    }
+
+    private async Task CancelConfirmation(string userchoice)
+    {
+        if (userchoice == "continueEdit")
+        {
+            //stop the cancel event and continue edit
+            ContinueEdit = true;
+        }
+        else
+        {
+            //cancel the edit
+            ContinueEdit = false;
+        }
+
+        //hide the dialog
+        DialogVisible = false;
+    }
+
+    protected override async Task OnInitializedAsync()
+    {
+        GridData = new List<SampleData>();
+
+        for (int i = 0; i < 50; i++)
```

Comment:
Start from 1. A Zero ID is equivalent to no ID and this creates problems with Create operations.
```suggestion
        for (int i = 1; i <= 50; i++)
```